src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -1375,6 +1376,11 @@ protected SurfaceType isWalkableSurface(@NotNull final BlockState blockState, fi
             return SurfaceType.NOT_PASSABLE;
         }
 
+        if (blockState.getBlock() == Blocks.WATER || (fluid != null && !fluid.isEmpty() && (fluid.getFluid() == Fluids.WATER || fluid.getFluid() == Fluids.FLOWING_WATER)))
```

Comment:
don't we have an isWater or so util?

Reply:
ye we do, though can't re-use the fluid state queried for the lava above then

Reply:
maybe break it off into a submethod then that is called, so we don't have different waterchecks all over the place potentially

Reply:
like a method that gets the fluidstate already,